src/scheduler/work.rs

```diff
@@ -4,9 +4,21 @@ use crate::vm::VMBinding;
 #[cfg(feature = "work_packet_stats")]
 use std::any::{type_name, TypeId};
 
+/// This defines a GC work packet which are assigned to the [`GCWorker`]s by the scheduler.
+/// Work packets carry payloads that indicate the work to be done. For example, a work packet may
+/// contain a pointer to a stack that must be scanned, or it may contain a large buffer of pointers
+/// that need to be traced, or it might contain a range of static variables to be scanned, etc. The size
+/// of the work packet will need to consider at least two points of tension: the work packet must be large
+/// enough to ensure that the costs of managing the work packets do not dominate, and the packet must be
+/// small enough that good load balancing is achieved.
 pub trait GCWork<VM: VMBinding>: 'static + Send {
     /// Define the work for this packet. However, this is not supposed to be called directly.
     /// Usually `do_work_with_stat()` should be used.
+    /// Only call this method if you would like to explicitly execute a work packet in the current context
+    /// (the given worker, and the current work packet). The statistics collected for executing the work are
+    /// counted into the current context. This is NOT the normal way to execute a work packet: most work packets
+    /// are polled and executed in the worker's main loop ([`GCWorker::run`]) using `do_work_with_stat`.
+    /// This method should only be called directly if you would like to explicitly execute a work packet under the current context.
```

Comment:
We never used the term "context" to refer to the worker and the work packet in other places, so it should be easier to understand if we don't use that word, or be more specific.

```suggestion
    ///
    /// Most work packets are polled and executed in the worker's main loop ([`GCWorker::run`])
    /// using `do_work_with_stat`.  If `do_work` is called directly during the execution of another
    /// work packet, bypassing `do_work_with_stat()`, this work packet will not be counted into the
    /// number of work packets executed, and the execution time of this work packet will be counted
    /// as part of the execution time of the other work packet.  Only call this method directly if
    /// this is what you intended.  But you should always consider putting the other work packet
    /// into a bucket so that other GC workers can execute it in parallel, unless the context-
    /// switching overhead is a problem.
```

Reply:
I updated the comments, and used most of your suggestions.